RNTester/NativeModuleExample/ScreenshotMacOS.mm

```diff
@@ -9,11 +9,10 @@
 #import <React/RCTUIManager.h>
 #import <React/RCTUtils.h>
 #import <ReactCommon/RCTTurboModuleManager.h>
-#import <ReactCommon/TurboModuleUtils.h>
 #import <NativeModules.h>
 #import <TurboModulesProvider.h>
 
-static NSImage* TakeScreenshot()
+static NSImage* TakeScreenshotAsImage()
```

Comment:
I only just now notice this, and it’s been a while since I’ve had to think about this, but I believe the Core Graphics image [created here below](https://github.com/microsoft/react-native-macos/blob/57015042c4d1daeb90ee15ed233ede865642cf67/RNTester/NativeModuleExample/ScreenshotMacOS.mm#L31) might be leaking.

In principle the memory rule is that memory allocated with an API that has the word “Create” or “Copy” needs to be released. With objc objects these are nowadays managed by ARC, but the objects returned from these C APIs are not automatically included unless you transfer ownership to ARC. (It might be best to verify if ARC is inserting the release statement by looking at the compilation result.)

In short, I think the code should be:

```objc
  CGImageRef windowImage = CGWindowListCreateImage(…);
  NSImage* image = [[NSImage alloc] initWithCGImage:windowImage size:[keyWindow frame].size];
  CGImageRelease(windowImage);
```

Perhaps @tom-un’s or @HeyImChris’ knowledge on the status quo of these rules is more up-to-date.

Reply:
Yes, good catch: the CGImageRef needs to be release via CGImageRelease or CFRelease

Reply:
Fixed

Reply:
All objc objects are managed by ARC, all Core Foundation objects are not (by default).

These are probably good places to start and get up to speed fast:
* https://developer.apple.com/library/archive/documentation/CoreFoundation/Conceptual/CFMemoryMgmt/Concepts/Ownership.html#//apple_ref/doc/uid/20001148-CJBEJBHH
* https://developer.apple.com/library/archive/releasenotes/ObjectiveC/RN-TransitioningToARC/Introduction/Introduction.html#//apple_ref/doc/uid/TP40011226-CH1-SW1